apps/settings/js/panels/full_developer_mode/panel.js

```diff
@@ -0,0 +1,45 @@
+define(function(require) {
+  'use strict';
+
+  var SettingsService = require('modules/settings_service');
+  var SettingsCache = require('modules/settings_cache');
+  var DialogService = require('modules/dialog_service');
+  var Panel = require('modules/panel');
+
+  return function() {
+    return Panel({
+      onInit: function(panel) {
+        this._panel = panel;
+        SettingsCache.getSettings((results) => {
+          var devMenuEnabled = !!results['developer.menu.enabled'];
+          if (devMenuEnabled) {
+            this._showFullDevModeWarning();
+          } else {
+            this._showEnableDevMenuWarning();
+          }
+        });
+      },
+      _close: function() {
+        SettingsService.navigate('home');
```

Comment:
In typical use cases a panel knows its parent panel, but there is no parent panel in this case. Therefore we need to have a virtual panel representing this case and which is `home`.
I agree `home` is not a good name. Maybe we should have a new API `close#SettingsService` that can be invoked in all panels.